tensorflow/contrib/cmake/CMakeLists.txt

```diff
@@ -39,8 +39,19 @@ if (NOT WIN32)
   # Threads: defines CMAKE_THREAD_LIBS_INIT and adds -pthread compile option
   # for targets that link ${CMAKE_THREAD_LIBS_INIT}.
   find_package (Threads)
+
+  option(tensorflow_PATH_STATIC_LIB "Additional library search path for libcudnn_static.a, libnccl_static.a, libculibos.a" /usr/local/cuda/lib64/)
```

Comment:
Do we need this option? For example, is there an underlying option that will cause `find_package(CUDA 8.0 REQUIRED)` to surface these same paths?

Reply:
Cudnn and nccl may be installed at a different path. (separated from Cuda)

Such issues get worse if users do not use popular Linux distro.

Reply:
Perhaps we should have two options, `tensorflow_PATH_CUDNN_STATIC_LIB` and `tensorflow_PATH_NCCL_STATIC_LIB`, with some common default?

Reply:
Ok. I'll add these two to override the value of ```tensorflow_PATH_STATIC_LIB```:
```
  option(tensorflow_PATH_CUDNN_STATIC_LIB "Override PATH_STATIC_LIB for libcudnn_static.a" ${tensorflow_PATH_STATIC_LIB})              
  option(tensorflow_PATH_NCCL_STATIC_LIB "Override PATH_STATIC_LIB for libnccl_static.a" ${tensorflow_PATH_STATIC_LIB}) 
```

Reply:
@mrry : I'm trying to extend this update to finish CMake + OSX testing w/ (NVIDIA) GPU.   I had started with CUDA 8.0 and cuDNN 6.0, but given recent activity here, it seems better to target master w/ CUDA 9.0 + cuDNN 7.0.  Since this is closely related, It seemed better to keep related discussion in this thread, but I can create a new issue if preferred.   The OS X CUDA installation does not come with `nccl` (see `find` output below for contents).  NVIDIA has a (login based) `nccl` download for v2.0, but it only has Linux libraries.  I see three other options to support OS X + GPU Builds:

1. https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/nccl (build from bundled source v1.3.5)
2. https://github.com/nvidia/nccl (build from source v1.?)
3. add `option(tensroflow_USE_NCCL "Use NCCL lib" ON)` and update code

Shall I add the bundled `nccl` (v1.3.5?) files to the CMake build?.  I see notes about migrating to v2.0 in commit comments, which seems to be closed source.  If that's the case, then it may be best to try to make any internal tensorflow `nccl` operations optional in order to support single GPU configurations to start with. 

`find /Developer/NVIDIA/CUDA-9.0/lib -name "*.a"`
```
/Developer/NVIDIA/CUDA-9.0/lib/libcublas_device.a
/Developer/NVIDIA/CUDA-9.0/lib/libcublas_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libcudadevrt.a
/Developer/NVIDIA/CUDA-9.0/lib/libcudart_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libcufft_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libcufftw_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libculibos.a
/Developer/NVIDIA/CUDA-9.0/lib/libcurand_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libcusolver_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libcusparse_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnppc_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnppial_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnppicc_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnppicom_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnppidei_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnppif_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnppig_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnppim_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnppist_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnppisu_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnppitc_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnpps_static.a
/Developer/NVIDIA/CUDA-9.0/lib/libnvgraph_static.a
```